File: fix-encoding.js
// fix-encoding.js - 用 Unicode 转义避免脚本自身的编码问题
const fs = require('fs');
const path = require('path');

const srcDir = 'C:\\Users\\a1478\\.openclaw-autoclaw\\workspace\\promotion-mvp\\frontend\\src';

// 用 Buffer 来存储替换映射，避免 JS 文件本身的编码问题
// 这些是从构建错误日志中提取的乱码模式
const replacements = JSON.parse(fs.readFileSync(path.join(__dirname, 'replacements.json'), 'utf8'));

const files = [
  'pages/HomeV2.tsx',
  'pages/ContentDetail.tsx',
  'pages/PublishV2.tsx',
  'pages/Settings.tsx',
  'pages/Tasks.tsx',
  'pages/TopicDetail.tsx',
  'pages/Promote.tsx',
  'components/CommentSheet.tsx',
];

for (const file of files) {
  const filePath = path.join(srcDir, file);
  if (!fs.existsSync(filePath)) {
    console.log(`SKIP: ${file} (not found)`);
    continue;
  }
  
  let content = fs.readFileSync(filePath, 'utf8');
  let changes = 0;
  
  for (const [from, to] of replacements) {
    while (content.includes(from)) {
      content = content.replace(from, to);
      changes++;
    }
  }
  
  fs.writeFileSync(filePath, content, 'utf8');
  console.log(`DONE: ${file} (${changes} replacements)`);
}

console.log('All files processed.');

File: fix_encoding.py
import os, sys, io
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')

src_root = r'C:\Users\a1478\.openclaw-autoclaw\workspace\promotion-mvp\frontend\src'

# All .tsx files that had BOM (are corrupted)
corrupted = []

for root, dirs, files in os.walk(src_root):
    for f in files:
        if not f.endswith('.tsx'):
            continue
        fp = os.path.join(root, f)
        with open(fp, 'rb') as fh:
            raw = fh.read()
        has_bom = raw[:3] == b'\xef\xbb\xbf'
        if has_bom:
            corrupted.append(fp)

print(f'Found {len(corrupted)} files with BOM:')
for fp in corrupted:
    print(f'  {os.path.relpath(fp, src_root)}')

fixed = 0
for fp in corrupted:
    with open(fp, 'rb') as fh:
        raw = fh.read()
    # Strip BOM
    raw = raw[3:]
    garbled = raw.decode('utf-8')
    
    # Method 1: latin-1 reversal (most common double-encoding)
    try:
        recovered = garbled.encode('latin-1').decode('utf-8')
        # Check if we got actual Chinese back
        chinese_count = sum(1 for c in recovered if '\u4e00' <= c <= '\u9fff')
        garbled_chinese = sum(1 for c in garbled if '\u4e00' <= c <= '\u9fff')
        if chinese_count > garbled_chinese or chinese_count > 0:
            with open(fp, 'wb') as fh:
                fh.write(recovered.encode('utf-8'))
            print(f'  FIXED (latin1): {os.path.basename(fp)} (Chinese chars: {garbled_chinese} -> {chinese_count})')
            fixed += 1
            continue
    except:
        pass
    
    # Method 2: cp1252 reversal
    try:
        recovered = garbled.encode('cp1252').decode('utf-8')
        chinese_count = sum(1 for c in recovered if '\u4e00' <= c <= '\u9fff')
        garbled_chinese = sum(1 for c in garbled if '\u4e00' <= c <= '\u9fff')
        if chinese_count > garbled_chinese or chinese_count > 0:
            with open(fp, 'wb') as fh:
                fh.write(recovered.encode('utf-8'))
            print(f'  FIXED (cp1252): {os.path.basename(fp)} (Chinese chars: {garbled_chinese} -> {chinese_count})')
            fixed += 1
            continue
    except:
        pass
    
    # Method 3: latin-1 bytes -> GBK decode
    try:
        bytes_iso = garbled.encode('latin-1')
        recovered = bytes_iso.decode('gbk')
        chinese_count = sum(1 for c in recovered if '\u4e00' <= c <= '\u9fff')
        if chinese_count > 0:
            with open(fp, 'wb') as fh:
                fh.write(recovered.encode('utf-8'))
            print(f'  FIXED (gbk): {os.path.basename(fp)} (Chinese chars: -> {chinese_count})')
            fixed += 1
            continue
    except:
        pass
    
    print(f'  FAILED: {os.path.basename(fp)} - no method worked. First 100 chars: {garbled[:100]}')

print(f'\nFixed: {fixed}/{len(corrupted)} files')
